src/NUnitFramework/framework/TestContext.cs

```diff
@@ -103,6 +104,17 @@ public string WorkDirectory
             }
         }
 
+        public static CompatibilityLogger Compatibility
+        {
+            get
+            {
+                var contextDictionary = CurrentContext._context;
+                return contextDictionary != null
+                    ? new CompatibilityLogger((TextWriter)contextDictionary["CompatibilityWriter"])
+                    : null;
```

Comment:
Why not `new CompatibilityLogger(Compatibility.Writer)` here?

Reply:
Why not `new CompatibilityLogger(Compatibility.Writer)` here?
Also should InvalidOperationException be thrown if you get a null `TextWriter`, meaning that collection has either not begun or has shut down? It's better than an ArgumentNullException.